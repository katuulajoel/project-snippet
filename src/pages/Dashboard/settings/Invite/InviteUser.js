import React from "react";
import TabBar from "./tabBar";

const Invite = () => (
  <>
    <TabBar />
    <h2>Invite user</h2>
  </>
);

export default Invite;
