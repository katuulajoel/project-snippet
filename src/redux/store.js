import { applyMiddleware, createStore } from "redux";
import reduxThunk from "redux-thunk";
import { composeWithDevTools } from "redux-devtools-extension";

import monitorReducersEnhancer from "../configs/enhancers/monitorReducer";
import loggerMiddleware from "../configs/middleware/logger";
import rootReducer from "./reducers";

export function configureStore(preloadedState = {}) {
  const middlewares = [loggerMiddleware, reduxThunk];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const enhancers = [middlewareEnhancer, monitorReducersEnhancer];
  const composedEnhancers = composeWithDevTools(...enhancers);

  const store = createStore(rootReducer, preloadedState, composedEnhancers);

  if (process.env.NODE_ENV !== "production" && module.hot) {
    module.hot.accept("./reducers", () => store.replaceReducer(rootReducer));
  }

  return store;
}

const store = configureStore();
export default store;
