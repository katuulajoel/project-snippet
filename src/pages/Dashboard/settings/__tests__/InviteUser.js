import React from "react";
import { Provider } from "react-redux";
import { cleanup, render, fireEvent } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import InviteUser from "../Invite/InviteUser";
import store from "../../../../redux/store";

afterEach(cleanup);

describe("InviteUser comp test", () => {
  it("InviteUser component snapshot", () => {
    const { asFragment } = render(
      <Provider store={store}>
        <Router>
          <InviteUser />
        </Router>
      </Provider>
    );

    expect(asFragment(<InviteUser />)).toMatchSnapshot();
  });

  it("Check 4 Labels render correctly on initial mount", () => {
    const { getAllByLabelText } = render(
      <Provider store={store}>
        <Router>
          <InviteUser />
        </Router>
      </Provider>
    );

    expect(getAllByLabelText("label")).toHaveLength(4);
  });

  it("Check User category Label does not render on initial mount", () => {
    const { queryByText } = render(
      <Provider store={store}>
        <Router>
          <InviteUser />
        </Router>
      </Provider>
    );

    expect(queryByText("User category")).toBeFalsy();
  });

  it("Check inputs value change", () => {
    const { queryByLabelText } = render(
      <Provider store={store}>
        <Router>
          <InviteUser />
        </Router>
      </Provider>
    );

    const emailInput = queryByLabelText("email-input");
    fireEvent.change(emailInput, { target: { value: "ms@gm.com" } });
    expect(emailInput.value).toBe("ms@gm.com");

    const firstNameInput = queryByLabelText("first_name-input");
    fireEvent.change(firstNameInput, { target: { value: "Kwame" } });
    expect(firstNameInput.value).toBe("Kwame");

    const lastNameInput = queryByLabelText("last_name-input");
    fireEvent.change(lastNameInput, { target: { value: "Mensah" } });
    expect(lastNameInput.value).toBe("Mensah");
  });

  it("Check form submit", () => {
    const { queryByLabelText, getByText } = render(
      <Provider store={store}>
        <Router>
          <InviteUser />
        </Router>
      </Provider>
    );

    const emailInput = queryByLabelText("email-input");
    fireEvent.change(emailInput, { target: { value: "ms@gm.com" } });
    expect(emailInput.value).toBe("ms@gm.com");

    const firstNameInput = queryByLabelText("first_name-input");
    fireEvent.change(firstNameInput, { target: { value: "Kwame" } });
    expect(firstNameInput.value).toBe("Kwame");

    const lastNameInput = queryByLabelText("last_name-input");
    fireEvent.change(lastNameInput, { target: { value: "Mensah" } });
    expect(lastNameInput.value).toBe("Mensah");

    const userTypeInput = queryByLabelText("last_name-input");
    fireEvent.change(userTypeInput, { target: { value: "1" } });
    expect(lastNameInput.value).toBe("1");

    fireEvent.click(getByText(/Send Invite/i));
  });
});
