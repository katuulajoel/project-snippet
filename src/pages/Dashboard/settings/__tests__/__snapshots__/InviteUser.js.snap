// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InviteUser comp test InviteUser component snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="sc-bdnxRM dNpxFo"
    >
      <div
        class="filter-links"
      >
        <a
          href="/settings/invite"
        >
          INVITE USERS
        </a>
        <a
          href="/settings/invite/create"
        >
          CREATE CLIENT
        </a>
        <a
          href="/settings/invite/pending"
        >
          View pending invites
        </a>
      </div>
    </div>
  </div>
  <div
    class="sc-eCApnc lwnPK"
  >
    <form>
      <div
        aria-label="label"
        class="row"
      >
        <div
          class="col-sm-12"
        >
          <div
            class="form-group"
          >
            <label
              aria-label="Email Address"
              class="control-label"
            >
              Email Address
              <span
                class="label-style"
              >
                *
              </span>
            </label>
            <div
              class="sc-jSFjdj dqrEkW"
              data-testid="input"
            >
              <input
                aria-label="email-input"
                class="form-control"
                data-testid=""
                id=""
                label="Label"
                name="email"
                placeholder="eg. bart@tunga.io"
                required=""
                type="input"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        aria-label="label"
        class="row"
      >
        <div
          class="col-sm-12"
        >
          <div
            class="form-group"
          >
            <label
              aria-label="First Name"
              class="control-label"
            >
              First Name
              <span
                class="label-style"
              >
                *
              </span>
            </label>
            <div
              class="sc-jSFjdj dqrEkW"
              data-testid="input"
            >
              <input
                aria-label="first_name-input"
                class="form-control"
                data-testid=""
                id=""
                label="Label"
                name="first_name"
                placeholder="Enter first name"
                required=""
                type="input"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        aria-label="label"
        class="row"
      >
        <div
          class="col-sm-12"
        >
          <div
            class="form-group"
          >
            <label
              aria-label="Last Name"
              class="control-label"
            >
              Last Name
              <span
                class="label-style"
              >
                *
              </span>
            </label>
            <div
              class="sc-jSFjdj dqrEkW"
              data-testid="input"
            >
              <input
                aria-label="last_name-input"
                class="form-control"
                data-testid=""
                id=""
                label="Label"
                name="last_name"
                placeholder="Enter last name"
                required=""
                type="input"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
      <div
        aria-label="label"
        class="row"
      >
        <div
          class="col-sm-12"
        >
          <div
            class="form-group"
          >
            <label
              aria-label="User type"
              class="control-label"
            >
              User type
              <span
                class="label-style"
              >
                *
              </span>
            </label>
            <select
              aria-label="type-input"
              class="form-control"
              name="type"
              required=""
            >
              <option
                value=""
              >
                Select type of user
              </option>
              <option
                value="1"
              >
                Developer
              </option>
            </select>
          </div>
        </div>
      </div>
      <div
        class="col-12"
      >
        <button
          class="btn btn-primary save"
        >
          Send Invite
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
